manager/state/raft/raft.go

```diff
@@ -1391,34 +1424,38 @@ func (n *Node) applyAddNode(cc raftpb.ConfChange) error {
 // from a member in the raft cluster, this removes a node
 // from the existing raft cluster
 func (n *Node) applyRemoveNode(cc raftpb.ConfChange) (err error) {
-	// If the node from where the remove is issued is
-	// a follower and the leader steps down, Campaign
-	// to be the leader.
-
-	if cc.NodeID == n.leader() && !n.isLeader() {
-		if err = n.Campaign(n.Ctx); err != nil {
-			return err
-		}
-	}
-
 	if cc.NodeID == n.Config.ID {
 		n.removeRaftFunc()
 
 		// wait the commit ack to be sent before closing connection
 		n.asyncTasks.Wait()
-
-		// if there are only 2 nodes in the cluster, and leader is leaving
-		// before closing the connection, leader has to ensure that follower gets
-		// noticed about this raft conf change commit. Otherwise, follower would
-		// assume there are still 2 nodes in the cluster and won't get elected
-		// into the leader by acquiring the majority (2 nodes)
-
-		// while n.asyncTasks.Wait() could be helpful in this case
-		// it's the best-effort strategy, because this send could be fail due to some errors (such as time limit exceeds)
-		// TODO(Runshen Zhu): use leadership transfer to solve this case, after vendoring raft 3.0+
 	}
-
-	return n.cluster.RemoveMember(cc.NodeID)
+	go func() {
+		l, err := n.Leader()
+		// if we're removing a leader - wait for leader change
```

Comment:
Does this work correctly on the old leader? I'd expect that other nodes would process the removal before doing an election, so the old leader would never know that a new leader was elected, and the loop below would time out.